Katana/Store.swift

```diff
@@ -201,6 +274,41 @@ open class Store<S: State, D: SideEffectDependencyContainer>: PartialStore<S> {
     }
   }
   
+  /**
+   Dispatches a `Dispatchable` item
+   
+   #### Threading
+   
+   The `Store` follows strict rules about the parallelism with which dispatched items are handled.
+   At the sime time, it tries to leverages as much as possible the modern multi-core systems that our
+   devices offer.
+   
+   When a `StateUpdater` is dispatched, the Store enqueues it in a serial and syncronous queue. This means that the Store
+   executes one update of the state at the time, following the order in which it has received them. This is done
+   to guarantee the predictability of the changes to the state and avoid any race condition. In general, using a syncronous
+   queue is never a big problem as any operation that goes in a `StateUpdater` is very lighweight.
+   
+   When it comes to `SideEffect` items, Katana will handle them in a parallel queue. A `SideEffect` is executed and considered
+   done when its body finishes to be executed. This menas that side effects are not guaranteed to be run in isolation, and you
+   should take into account the fact that multiple side effects can run at the time time. This decision has been taken to greately
+   improve the performances of the system of out the box. Overall, this should not be a problem as you cannot really change
+   the state of the system (that is, the store's state) without dispatching a `StateUpdater`.
+   
+   This verison of the store keeps the support for `Action` items. Since actions both update the state and executes side effects,
+   they are managed in the very same, serial and syncronous, queue of the `StateUpdater`. You are encouraged to move away from
+   actions as soon as possible.
+   
+   #### Promise Resolution
+   
+   When it comes to `StateUpdater`, the promise is resolved when the state is updated. For `SideEffect`,
```

Comment:
hoqw does it behave with the old `AsyncAction`?

Reply:
`AsyncAction` is just a wrapper around normal actions, so it works exactly as before